pyproject.toml

```diff
@@ -0,0 +1,105 @@
+[build-system]
+requires = [
+    "setuptools >= 40.6.0",
+    "setuptools_scm>=6.2",
+]
+build-backend = "setuptools.build_meta"
+
+[project]
+authors = [
+    {name = "Pyfive Development Team", email = "jjhelmus@gmail.com"}
+]
+classifiers = [
+    "Development Status :: 5 - Production/Stable",
+    "Environment :: Console",
+    "Intended Audience :: Developers",
+    "Intended Audience :: Science/Research",
+    "License :: OSI Approved :: BSD License",
+    "Natural Language :: English",
+    "Operating System :: POSIX :: OS Independent",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Programming Language :: Python :: 3.12",
+    "Programming Language :: Python :: 3.13",
+    "Topic :: Scientific/Engineering",
+    "Topic :: Scientific/Engineering :: Atmospheric Science",
+    "Topic :: Scientific/Engineering :: Physics",
+]
+dynamic = [
+    "readme",
+    "version",
+]
+dependencies = [
+    "numpy",
+    "pip!=21.3",
```

Comment:
pip should not be included as a package dependency. It is a front end that can be used to install pyfive but it is not needed at runtime.

Reply:
I tend to agree with you on this one, but the problem is that it becomes murky which `pip` is used where and by what - we should avoid using `pip` default from PyPI when working with conda packages. If you are 100% that this package will only need `numpy` as a build dependency, then we can remove `pip` here, but if Pyfive will start adding a number of new build dependencies from conda-forge then we should most deffo use the conda-forge `pip` (and have it as a build/install dependency)

Reply:
I'm a huge fan of conda-forge, but pyfive is a pure-Python package and I'm hoping it won't need to be built (or have build dependencies)

Reply:
sure, but other of its (possible future) dependencies may eg `netCDF4` - getting those off PyPI is a bit of a pain :grin:

Reply:
With this listed as a dependency `pip` will be installed into the environment from PyPI by the `pip` installed by `conda` but only if the upgrade strategy is eager (not the default) or something else required a different version of `pip`. In the current run the `conda` installed version is retained.  I can't see any case where this would be needed, the packages listed here are installed by pip itself (or another PEP 517 front end).

Reply:
> sure, but other of its (possible future) dependencies may eg `netCDF4` - getting those off PyPI is a bit of a pain 😁

I hope, `pyfive` will never depend on `netCDF4` as this depends on `netcdf-c` which depends on `hdf5` which counters the efforts of `pyfive`, IMHO.

Reply:
ah no way, that was a somewhat anecdotal example :smiley: Let me just remobe `pip` from the conda env, then - so that `numpy` is the only one left as install dep :+1: